pkg/prometheus/server/operator.go

```diff
@@ -351,10 +368,19 @@ func (c *Operator) waitForCacheSync(ctx context.Context) error {
 		{"PodMonitor", c.pmonInfs},
 		{"PrometheusRule", c.ruleInfs},
 		{"Probe", c.probeInfs},
+		{"ScrapeConfig", c.sconInfs},
 		{"ConfigMap", c.cmapInfs},
 		{"Secret", c.secrInfs},
 		{"StatefulSet", c.ssetInfs},
 	} {
+		// Skipping ScrapeConfigs if we don't have access to the CRD or if the CRD is not installed
+		if infs.name == "ScrapeConfig" {
+			if ok, err := c.isScrapeConfigInstalled(ctx); !ok || err != nil {
```

Comment:
I'm not confident we want this condition like this, mainly because it will hide the error and it will make it hard to debug why the ScrapeConfig informer is not being started

Reply:
Sounds good to me! Do we want to wait for #5476 before merging this one or are we fine with a followup?

Reply:
I'd rather wait for #5476. I know that @slashpai is actively on it :)